File: src/main/java/RemoveDuplicatesfromSortedList/haierbopu/RemoveDuplicatesfromSortedList.py
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def deleteDuplicates(self, head: ListNode) -> ListNode:
        if not head:
            return head
        prev = head
        curr = head.next
        while curr:
            if curr.val == prev.val:
                prev.next = curr.next
            else:
                prev = curr
            curr = curr.next
        return head

File: src/main/java/RemoveDuplicatesfromSortedList/haierbopu/RemoveDuplicatesfromSortedList.java
class Solution {
  public class ListNode {
    int val;
    ListNode next;
    ListNode(int x) { val = x; }
  }
    public ListNode deleteDuplicates(ListNode head) {
        if (head == null) return head;
        ListNode prev = head;
        ListNode curr = head.next;
        while (curr != null){
            if (curr.val == prev.val){
                prev.next = curr.next;
            }
            else{
                prev = curr;
            }
            curr = curr.next;
        }
        return head;
    }
    public static void main(String []args) {
        System.out.println("RemoveDuplicatesfromSortedList");
    }
}
